packages/php/blueprint/src/ExportSchema.php

```diff
@@ -100,19 +102,24 @@ function ( $exporter ) {
 			}
 		}
 
+		$logger = new Logger();
+		$logger->start_export( $exporters );
+
 		foreach ( $exporters as $exporter ) {
-			$this->publish( 'onBeforeExport', $exporter );
-			$step = $exporter->export();
+			try {
+				$this->publish( 'onBeforeExport', $exporter );
+				$step = $exporter->export();
+				$this->add_result_to_schema( $schema, $step );
 
-			if ( is_array( $step ) ) {
-				foreach ( $step as $_step ) {
-					$schema['steps'][] = $_step->get_json_array();
-				}
-			} else {
-				$schema['steps'][] = $step->get_json_array();
+			} catch ( \Throwable $e ) {
+				$step_name = $exporter instanceof HasAlias ? $exporter->get_alias() : $exporter->get_step_name();
+				$logger->export_step_failed( $step_name, $e );
```

Comment:
May be nice to just pass the `$exporter` to the `step_failed` and also move the `$step_name` logic to the Logger.
This is small though, so not a blocker to merge.